{"schemaVersion":{"minor":3,"major":0,"patch":0},"metadata":{"role":"collectionGroup","roleHeading":"API Collection","modules":[{"name":"Amplify"}],"title":"AmplifyError Implementations"},"sections":[],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/AmplifyError-Implementations"},"hierarchy":{"paths":[["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError"]]},"kind":"article","topicSections":[{"identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/init(errorDescription:recoverySuggestion:error:)"],"title":"Initializers","generated":true,"anchor":"Initializers"},{"identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/errorDescription","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/isOperationCancelledError","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/recoverySuggestion","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/underlyingError"],"generated":true,"title":"Instance Properties","anchor":"Instance-Properties"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/amplify\/huberror\/amplifyerror-implementations"]}],"references":{"doc://amplify-swift.Amplify/documentation/Amplify/HubError/underlyingError":{"url":"\/documentation\/amplify\/huberror\/underlyingerror","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"text":"underlyingError","kind":"identifier"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"text":"?","kind":"text"}],"kind":"symbol","type":"topic","abstract":[{"type":"text","text":"The underlying error that caused the error condition"}],"title":"underlyingError","role":"symbol","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/underlyingError"},"doc://amplify-swift.Amplify/documentation/Amplify/HubError":{"url":"\/documentation\/amplify\/huberror","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"HubError","kind":"identifier"}],"kind":"symbol","type":"topic","abstract":[{"type":"text","text":"Errors associated with configuring and inspecting Amplify Categories"}],"title":"HubError","role":"symbol","navigatorTitle":[{"text":"HubError","kind":"identifier"}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError"},"doc://amplify-swift.Amplify/documentation/Amplify":{"url":"\/documentation\/amplify","type":"topic","title":"Amplify","role":"collection","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","kind":"symbol","abstract":[]},"doc://amplify-swift.Amplify/documentation/Amplify/HubError/errorDescription":{"abstract":[{"text":"A localized message describing what error occurred.","type":"text"}],"type":"topic","title":"errorDescription","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/errorDescription","url":"\/documentation\/amplify\/huberror\/errordescription","role":"symbol","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"errorDescription","kind":"identifier"},{"kind":"text","text":": "},{"preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier","text":"ErrorDescription"}],"kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/HubError/recoverySuggestion":{"role":"symbol","title":"recoverySuggestion","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/recoverySuggestion","kind":"symbol","abstract":[{"text":"A localized message describing how one might recover from the failure.","type":"text"}],"type":"topic","fragments":[{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"text":"recoverySuggestion","kind":"identifier"},{"text":": ","kind":"text"},{"text":"RecoverySuggestion","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona"}],"url":"\/documentation\/amplify\/huberror\/recoverysuggestion"},"doc://amplify-swift.Amplify/documentation/Amplify/HubError/isOperationCancelledError":{"abstract":[],"type":"topic","title":"isOperationCancelledError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/isOperationCancelledError","url":"\/documentation\/amplify\/huberror\/isoperationcancellederror","role":"symbol","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"isOperationCancelledError","kind":"identifier"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}],"kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/HubError/init(errorDescription:recoverySuggestion:error:)":{"role":"symbol","title":"init(errorDescription:recoverySuggestion:error:)","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/HubError\/init(errorDescription:recoverySuggestion:error:)","kind":"symbol","abstract":[{"type":"text","text":"AmplifyErrors must be able to be initialized from an underlying error. If an AmplifyError is created"},{"type":"text","text":" "},{"type":"text","text":"with this initializer, it must store the underlying error in the "},{"type":"codeVoice","code":"underlyingError"},{"text":" property so it can be","type":"text"},{"text":" ","type":"text"},{"text":"inspected later.","type":"text"}],"type":"topic","fragments":[{"text":"init","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"errorDescription"},{"text":": ","kind":"text"},{"text":"ErrorDescription","preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"kind":"externalParam","text":"recoverySuggestion"},{"kind":"text","text":": "},{"text":"RecoverySuggestion","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona"},{"kind":"text","text":", "},{"text":"error","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"kind":"text","text":")"}],"url":"\/documentation\/amplify\/huberror\/init(errordescription:recoverysuggestion:error:)"}}}